src/sentry/search/django/backend.py

```diff
@@ -8,275 +8,397 @@
+    return '"{}"."{}"'.format(*[
+        model._meta.db_table,
+        model._meta.get_field_by_name(field)[0].column,
+    ])
+
+
+sort_strategies = {
+    # sort_by -> Tuple[
+    #   Paginator,
+    #   String: QuerySet order_by parameter
+    # ]
+    'priority': (Paginator, '-score'),
+    'date': (DateTimePaginator, '-last_seen'),
+    'new': (DateTimePaginator, '-first_seen'),
+    'freq': (Paginator, '-times_seen'),
+}
+
+environment_sort_strategies = {
+    # sort_by -> Tuple[
+    #   String: SQL expression to generate sort value (of type T, used below),
+    #   Function[T] -> int: function for converting sort value to cursor value),
+    # ]
+    'priority': ('log(times_seen) * 600 + last_seen::abstime::int', int),
+    'date': ('last_seen', lambda score: int(to_timestamp(score) * 1000)),
+    'new': ('first_seen', lambda score: int(to_timestamp(score) * 1000)),
+    'freq': ('times_seen', int),
+}
+
+
+def get_sort_clause(sort_by):
+    engine = get_db_engine('default')
+    if engine.startswith('sqlite'):
+        return SQLITE_SORT_CLAUSES[sort_by]
+    elif engine.startswith('mysql'):
+        return MYSQL_SORT_CLAUSES[sort_by]
+    elif engine.startswith('oracle'):
+        return ORACLE_SORT_CLAUSES[sort_by]
+    elif engine in MSSQL_ENGINES:
+        return MSSQL_SORT_CLAUSES[sort_by]
+    else:
+        return SORT_CLAUSES[sort_by]
 
-        if bookmarked_by:
-            queryset = queryset.filter(
-                bookmark_set__project=project,
-                bookmark_set__user=bookmarked_by,
-            )
 
-        if assigned_to:
-            queryset = queryset.filter(
-                assignee_set__project=project,
-                assignee_set__user=assigned_to,
-            )
-        elif unassigned in (True, False):
-            queryset = queryset.filter(
-                assignee_set__isnull=unassigned,
-            )
+class DjangoSearchBackend(SearchBackend):
+    def query(self, project, tags=None, environment=None, sort_by='date', limit=100,
+              cursor=None, count_hits=False, paginator_options=None, **parameters):
+        from sentry.models import Environment, Group, GroupEnvironment, GroupStatus, GroupSubscription, Release
 
-        if subscribed_by is not None:
-            queryset = queryset.filter(
-                id__in=GroupSubscription.objects.filter(
-                    project=project,
-                    user=subscribed_by,
-                    is_active=True,
-                ).values_list('group'),
-            )
+        if paginator_options is None:
+            paginator_options = {}
 
-        if first_release:
-            if first_release is EMPTY:
-                return queryset.none()
-            queryset = queryset.filter(
-                first_release__organization_id=project.organization_id,
-                first_release__version=first_release,
-            )
+        if tags is None:
+            tags = {}
+
+        group_queryset = QuerySetBuilder({
```

Comment:
These are all of the query parameters that apply that aren't derived from event data.